lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -188,7 +188,7 @@ def parse_relationship(assoc_name, assoc_data, options)
             end
 
           polymorphic = (options[:polymorphic] || []).include?(assoc_name.to_sym)
-          hash["#{prefix_key}_type".to_sym] = assoc_data[:type] if polymorphic
+          hash["#{prefix_key}_type".to_sym] = assoc_data.try(:[], :type) if polymorphic
```

Comment:
in Ember, if an association isn't set: upon saving, the data for that association is just straight up `nil`